js/script.js

```diff
@@ -0,0 +1,456 @@
+  const curPos = textarea.selectionStart;
+  const start = textarea.selectionStart;
+  const end = textarea.selectionEnd;
+  textarea.focus();
+
+  switch (id) {
+    case 'Backspace':
+      if (end === start && start === 0) {
+        break;
+      }
+      if (end > start) {
+        textarea.setRangeText('');
+        break;
+      }
+      textarea.value = textarea.value.slice(0, curPos - 1) + textarea.value.slice(curPos);
+      textarea.selectionStart = curPos - 1;
+      textarea.selectionEnd = curPos - 1;
+      break;
+    case 'Tab':
+      textarea.setRangeText('    ');
+      textarea.selectionStart = curPos + 4;
+      break;
+    case 'Delete':
+      if (end > start) {
+        textarea.setRangeText('');
+        break;
+      }
+      textarea.value = textarea.value.slice(0, curPos) + textarea.value.slice(curPos + 1);
+      textarea.selectionStart = curPos;
+      textarea.selectionEnd = curPos;
+      break;
+    case 'CapsLock':
+      if (!keyPressedFlag) {
+        isCaps = !isCaps;
+        document.querySelector('#CapsLock').classList.toggle('keyboard__key--active');
+        changeKeyboardLayout('caseChangeOnCaps');
+        keyPressedFlag = true;
+      }
+      break;
+    case 'Enter':
+      textarea.setRangeText('\n');
+      textarea.selectionStart = curPos + 1;
+      break;
+    case 'ShiftLeft':
+    case 'ShiftRight':
+      isShiftPressed = true;
+      break;
+    case 'AltLeft':
+      isAltPressed = true;
+      break;
+    case 'ControlLeft':
+      isCtrlPressed = true;
+      break;
+    case 'ControlRight':
+    case 'AltRight':
+    case 'MetaLeft':
+      break;
+    default:
+      textarea.setRangeText(input);
+      textarea.selectionStart = curPos + 1;
+      break;
+  }
+
+  if (isAltPressed && isCtrlPressed && !keyPressedFlag) {
+    isEng = !isEng;
+    changeKeyboardLayout('langChange');
+    localStorage.setItem('isEng', isEng);
+    keyPressedFlag = true;
+  }
+
+  if (isShiftPressed && !keyPressedFlag) {
+    changeKeyboardLayout('caseChangeOnShift');
+    keyPressedFlag = true;
+  }
+}
+
+function keyUnpressHandler(id) {
+  const textarea = document.querySelector('.textarea');
+  textarea.focus();
+  switch (id) {
+    case 'ShiftLeft':
+    case 'ShiftRight':
+      isShiftPressed = false;
+      changeKeyboardLayout('caseChangeAfterShift');
+      break;
+    case 'AltLeft':
+      isAltPressed = false;
+      break;
+    case 'ControlLeft':
+      isCtrlPressed = false;
+      break;
+    default:
+      break;
+  }
+
+  keyPressedFlag = false;
+}
+
+function keyDownHandler(e) {
+  e.preventDefault();
+  if (isKeyOnVirtualKeyboard(e.code)) {
+    document.querySelector(`#${e.code}`).classList.add('keyboard__key--pressed');
+
+    const id = document.querySelector(`#${e.code}`).getAttribute('id');
+    const input = document.querySelector(`#${e.code}`).textContent;
+    keypressHandler(id, input);
+  }
+}
+
+function keyUpHandler(e) {
```

Comment:
Use destructuring for e param (keyUpHandler({code}) ) and use it inside function.

Reply:
function f(props) {
  const {arr} = props;
}
props.arr